src/Controller/Produtos/ProdutoApiController.php

```diff
@@ -18,4 +18,17 @@ public function listarProdutos(Request $request, ProdutoRepository $produtoRepos
 
         return $this->json($produtos);
     }
+
+    #[Route('/api/produtos/{id}', name: 'api_produto', methods: ['GET'])]
+    public function buscarProduto($id, ProdutoRepository $produtoRepository): JsonResponse
+    {
+        $produto = $produtoRepository->find($id);
+        if (!$produto) {
+            return new JsonResponse([
+                'error' => 'Produto não encontrado',
+            ], JsonResponse::HTTP_NOT_FOUND);
+        }

```

Comment:
faz igual fizemos sabado, cria um método chamado, buscarPorIdOuFalhar e lá você olha se existe o produto e lança exception...
cria uma chamada ProdutoNaoEncontradoException

```suggestion
        $this->json($produtoRepository->buscarPorIdOuFalhar($id));
```

Reply:
tem remover o return da linha 32 em...

Reply:
Conforme a gente conversou, eu irei realizar as alterações, mas para ganhar tempo irei realizar as tarefas mais importes e depois eu realizo as alterações sugeridas

Reply:
no caso apenas essa alteraçãoa aqui era para pular @GabrielCirqueira e não todas as alterações, ela para fazer as demais...